test/router/path_trie_tail_test.dart

```diff
@@ -0,0 +1,265 @@
+import 'package:relic/src/router/normalized_path.dart';
+import 'package:relic/src/router/path_trie.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('PathTrie Tail (**) Matching', () {
+    late PathTrie<int> trie;
+
+    setUp(() {
+      trie = PathTrie<int>();
+    });
+
+    test(
+        'Given a trie with path /static/**, '
+        'when /static/css/style.css is looked up, '
+        'then it matches with correct value and remaining path', () {
+      trie.add(NormalizedPath('/static/**'), 1);
+      final result = trie.lookup(NormalizedPath('/static/css/style.css'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, isEmpty);
+      expect(result.matched.path, '/static');
+      expect(result.remaining.path, '/css/style.css');
```

Comment:
Where does the requirement come from that `path` above (and in old code) and `remaining` start with `/`?

I would think in both cases they are relative paths, as the "route" (in practice) does not know what it's parent / mount point will be, and the remaining path / sub segment is definitely relative to it's parent.

I think the static file serving is a good example.
Does Dart have a trasversal safe API where you could do something like `Directory('/data/static').resolveSafe(userInputFilePath)`? (Go added this recently.)

Then I would maybe try to see to give a value that matches this and can be nicely forwarded.

Or maybe my hunch is wrong here, and even "relative" paths have a benefit of starting with `/`.

Reply:
Maybe I'm wrong, but don't the path component of an url always start with `/`?

Reply:
> Maybe I'm wrong, but don't the path component of an url always start with /?

I think there is an explicit distinction between absolute and relative URIs, and I just wondered whether by definition not all of these in there would be relative.